whittle/training_strategies/ats.py

```diff
@@ -48,7 +48,8 @@ def __call__(self, model, inputs, outputs, **kwargs):
                     loss = self.kd_loss(y_hat, outputs, y_supernet)
                 else:
                     loss = self.loss_function(y_hat, outputs)
-                loss.backward()
+                loss *= scale_loss
+                loss.backward() if self.fabric is None else self.fabric.backward(loss)
```

Comment:
If we use litgpt scripts, isn't self.fabric always set?

Reply:
yes, for some projects (e.g the plm pruning project) we don't use fabric.